src/main/java/sdu/se9/tv2/management/system/persistence/PersistenceCredit.java

```diff
@@ -21,46 +23,49 @@ public static PersistenceCredit getInstance() {
         return instance;
     }
 
-    private Persistence persistence = new Persistence("credit.json");
-
-    private int lastID = -1;
-
-    private ArrayList<Credit> credits = new ArrayList<Credit>();
-
-    private PersistenceCredit() {
-        this.read();
-    }
+    private PersistenceCredit() {}
 
     @Override
-    public Credit createCredit(int programID, int personID, String roleName) throws DuplicateRoleNameException {
-        if (this.getCredit(programID, roleName) != null) {
-            throw new DuplicateRoleNameException("Role already exists");
-        }
+    public Credit createCredit(int programID, int personID, String roleName) throws DuplicateRoleNameException, SQLException {
 
-        lastID++;
-        Credit newCredit = new Credit(lastID, programID, personID, roleName);
-        credits.add(newCredit);
+        Connection connection = PersistenceDatabaseHelper.getConnection();
 
-        // Save changes to file
+            if (this.getCredit(programID, roleName) != null) {
+                throw new DuplicateRoleNameException("Role already exists");
+            }
 
-        this.write();
+            PreparedStatement stmt = connection.prepareStatement("INSERT INTO Credit(programID, personID, roleName) VALUES(?,?,?) ", Statement.RETURN_GENERATED_KEYS);
+            stmt.setInt(1,  programID);
+            stmt.setInt(2,  personID);
+            stmt.setString(3,  roleName);
+            stmt.execute();
+            ResultSet rs = stmt.getGeneratedKeys();
+            int i = 0;
+            if (rs.next()) {
+                i = rs.getInt(1);
+            }
 
-        return newCredit;
+            System.out.println("You did it!");
+            System.out.println(i);
+            return new Credit(i, programID, personID, roleName);
+        // Save changes to file
     }
 
     @Override
-    public ArrayList<Credit> getCredits(int programID) {
-        ArrayList<Credit> result = new ArrayList<Credit>();
+    public ArrayList<Credit> getCredits(int programID) throws SQLException {
+        Connection connection = PersistenceDatabaseHelper.getConnection();
 
-        for (int i = 0; i < this.credits.size(); i++) {
-            Credit element = this.credits.get(i);
 
-            if (element.getProgramID() == programID) {
-                result.add(element);
+            PreparedStatement stmt = connection.prepareStatement("SELECT * FROM Credit WHERE programID =" + programID);
+            ResultSet result = stmt.executeQuery();
+            int rowcount = 0;
+            ArrayList<Credit> returnValue = new ArrayList<>();
+            while (result.next()){
+                returnValue.add(new Credit(result.getInt(1), result.getInt(2), result.getInt(3), result.getString(4)));
```

Comment:
Brug navnet på kolonnerne i stedet for index for kolonnen. Hvis af en eller anden grund id ikke er det første i tabellen så kommer der til at ske en fejl, så for at sørge for at det ikke skulle ske så brug i stedet navnet på kolonnen:

```java
new Credit(result.getInt("id"), result.getInt("programID"), result.getInt("personID"), result.getString("role"));
```

Koden over virker sikkert ikke, men det er et eksempel på hvordan i skal gøre det.